### PR TITLE
Czech cheatsheet

### PR DESCRIPTION
**Pull request by comatory**
**What does your Instant Answer do?**
Displays common phrases in Czech language with appropriate English translations.

**What is the data source for your Instant Answer? (Provide a link if possible)**
There is a data source pointing to URL at Local Lingo but I'm using it to point people to other words and phrases. Primarily my source is my native tongue so I'm translating it. Second, I used German cheat sheet as template (`share/goodie/cheat_sheets/json/german.json`)

**Why did you choose this data source?**
It is not overwhelmingly long and shows only the important stuff. It's instant answer so logically, for further info you'd go to Google translate or somewhere else.

**Are there any other alternative (better) data sources?**
Can't think of any.

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
Tourists or people moving in to Czech Republic. Good for beginners as well.

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
Yes -> [IA Page](https://duck.co/ideas/idea/6412/czech-phrases-cheat-sheet)

**Are you having any problems? Do you need our help with anything?**
Yes. I cannot make DDG to trigger my IA with the aliases specified in source file (see line 9 in `czech.json`). The only trigger that works is "czech cheat sheet".

**Where did you hear about DuckDuckHack? (For first time contributors)**
On your DDG website.

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
ddh-screenshot

-----
IA Page: https://duck.co/ia/view/czech_cheat_sheet

### CODE REVIEW

**Comment by moollaza** [comment]
@comatory thanks for your contribution!

Can you please fill out the [pull request template](https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/pull_request_template_cheat_sheet.md) and edit your initial comment. Once completed, we'll be sure to review your PR and give you some feedback.

Thanks!

**Comment by comatory** [comment]
@moollaza thanks I didn't know about that request template. I guess I didn't read DDH enough :) I updated my comment.

**Comment by moollaza** [comment]
>Yes. I cannot make DDG to trigger my IA with the aliases specified in source file (see line 9 in czech.json). The only trigger that works is "czech cheat sheet".

@comatory what were the queries you tried? The aliases are only for the filename, so you still need the term "cheat sheet" in the query to trigger this.

e.g. "czech phrases cheat sheet"

File: share/goodie/cheat_sheets/json/czech.json [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+      "Travelling",
+      "Etiquette",
+      "Dinner on the Town"
+   ],
+   "description" : "Basic Czech words and phrases",
+   "sections" : {
+      "Etiquette" : [
+         {
+            "val" : "Dobrou noc",
+            "key" : "Good Night"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/czech.json]: We can lowercase `night`, `afternoon`, `morning` and `evening`

File: share/goodie/cheat_sheets/json/czech.json [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+         },
+         {
+            "val" : "Nemáte zač",
+            "key" : "You're welcome"
+         }
+      ],
+      "Getting Help" : [
+         {
+            "val" : "Zavolejte policii!",
+            "key" : "Call the Police!"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/czech.json]: let's lowercase `police`

**Comment by comatory** [comment]
@moollaza Ok Zahir I fixed the strings. As for the triggers you were right - I was trying general queries like `czech phrases` without adding `cheat sheet` to it.
I'm wondering is there an easy way to use aliases as queries without `cheat sheet` in it? I think it would be nice if this cheat sheet could be triggered by terms `czech phrases` or `basic czech`.

**Comment by moollaza** [comment]
>I'm wondering is there an easy way to use aliases as queries without cheat sheet in it? I think it would be nice if this cheat sheet could be triggered by terms czech phrases or basic czech.

@comatory Not right now, but we're currently in investigating other ways of determining when it's appropriate to show a cheat sheet based on the name/description/content. For now it's fine as-is.

**Comment by moollaza** [comment]
@comatory it's live on beta.duckduckgoc.com: https://beta.duckduckgo.com/?q=czech%20cheat%20sheet&ia=cheatsheet

I've made a task to ask the community to sign-off on it :+1:

**Comment by comatory** [comment]
@moollaza cool. thanks!

**Comment by MikkCZ** [comment]
"Good afternoon" should be "Dobré odpoledne"
"Thank you very much!" "Děkuji vám mnohokrát!"
"Repeat, please." "Zopakujte to, prosím."
"How do you say "X" in Czech?" "Jak se česky řekne "X"?" 
"Do you speak English?" "Mluvíte anglicky?"
"Excuse me!" "Promiňte!"

**Comment by IAmNotImportant** [comment]
I think that "A glass of water please." should be translated as "Prosím sklenici vody." (current is "Sklenku vody prosím"). "Sklenice" is much more common, it sounds better as well.

**Comment by comatory** [comment]
@MikkCZ @IAmNotImportant good points. Will fix it now.

**Comment by comatory** [comment]
Fixed. Looking at that some translations were not so perfect.

**Comment by MikkCZ** [comment]
Is there any stable Czech l10n community from which these translations originate?

@comatory seems you've omitted the suggestion from @IAmNotImportant, I do not see it in the commit. ;)

**Comment by comatory** [comment]
@MikkCZ no they are my own translation, I am no professional :) However I was inspired by the german cheat sheet and I'm using the same phrases. @IAmNotImportant 's suggestion is now implemented.

**Comment by moollaza** [comment]
@comatory @MikkCZ @IAmNotImportant thank you ***very*** much for the collaboration and improvements :+1: 

The updates are live on Beta: https://beta.duckduckgo.com/?q=czech+cheat+sheet&ia=cheatsheet&iax=1

If we're all in agreement that it's finished I'd be happy to merge :smiley:

**Comment by MikkCZ** [comment]
For me it's OK. :)

**Comment by comatory** [comment]
@MikkCZ @IAmNotImportant thanks again for pointing out the mistakes.

**Comment by moollaza** [comment]
Sorry for the delay guys! This LGTM :+1:

**Comment by moollaza** [comment]
@comatory congrats! It's live: https://duckduckgo.com/?q=czech+cheat+sheet&ia=cheatsheet

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

**Pull request merged**